infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServerCache.java

```diff
@@ -167,15 +126,25 @@ public long size() {
      * @param newTail        global tail
      */
     public void put(ConflictTxStream conflictStream, long newTail) {
-        conflictCache.put(conflictStream, newTail);
+        if (cacheHash.size() == cacheSize) {
+            ConflictTxStreamEntry entry = cacheEntries.peek();
+            invalidateUpTo(firstAddress() + 1);
```

Comment:
At some point the cache will fill up and every put will translate into an eviction first, instead of having this code being invoked on every put, you can just do a  "bulk" eviction which is a % of the cache. For example, evict 25% of the cache whenever evict is invoked.

Reply:
But the eviction (even one by one) is very cheap. I wouldn't make the eviction logic more complex. I would prefer the current approach.
But up to you guys